src/components/header/SearchBar.js

```diff
@@ -49,48 +70,145 @@ const styles = theme => ({
     [theme.breakpoints.up('md')]: {
       width: 200,
     },
+    [theme.breakpoints.down('xs')]: {
+      paddingLeft: theme.spacing.unit * 4,
+    },
+  },
+  menuStyle: {
+    zIndex: 1,
+    minWidth: '200px',
+    maxHeight: '200px',
+    overflow: 'auto',
   },
 });
 
-export class SearchBar extends Component {
+class SearchBar extends Component {
   constructor(props) {
     super(props);
     this.state = {
       searchValue: '',
+      open: false,
+      shownEnv: modDisplay,
     };
     this.handleSearch = this.handleSearch.bind(this);
     this.delayedSearch = debounce(this.props.searchEnvironments, 500);
     this.delayedReset = debounce(this.props.resetSearchedEnvironments, 500);
   }
 
+  handleExpandMore = () => {
+    var showMore = this.state.shownEnv;
+    showMore += modDisplay;
+    this.setState({ shownEnv: showMore });
+  };
+
+  handleExpandLess = () => {
+    var showLess = this.state.shownEnv;
+    showLess -= modDisplay;
+    this.setState({ shownEnv: showLess });
+  };
+
   handleSearch(event) {
     // TODO: LOADING and FAILURE and X-out button
     event.preventDefault();
-    this.setState({ searchValue: event.target.value });
+    this.setState({ searchValue: event.target.value, open: true });
     if (event.target.value.trim().length >= 3) {
       this.delayedSearch(event.target.value);
     } else {
       this.delayedReset();
     }
   }
+  handleClose = () => {
+    this.setState({ open: false });
+  };
 
   render() {
-    const { searchValue } = this.state;
+    const { searchValue, open } = this.state;
     const { classes } = this.props;
+    var environments = this.props.searchedEnvironments
+      ? this.props.searchedEnvironments
+      : this.props.environments;
+    const all = true ? this.state.shownEnv >= environments.length : false;
+    const none = true ? this.state.shownEnv <= modDisplay : false;
+    environments = environments.slice(0, this.state.shownEnv);
```

Comment:
possibly give `environments` to `visibleEnvironments` so you can use `const` instead of `var`. Also it's a bit more descriptive.

Reply:
done